tests/apps/pubsub-subscriber/app.go

```diff
@@ -0,0 +1,178 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	appPort = 3000
+	pubsubA = "pubsub-a-topic"
+	pubsubB = "pubsub-b-topic"
+)
+
+type appResponse struct {
+	Message   string `json:"message,omitempty"`
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+}
+
+type topicsList struct {
+	Topic []string
+}
+
+type receivedMessagesResponse struct {
+	ReceivedByTopicA []string `json:"pubsub-a-topic"`
+	ReceivedByTopicB []string `json:"pubsub-b-topic"`
+}
+
+var receivedMessagesA []string
+var receivedMessagesB []string
+
+var lock sync.Mutex
+
+// indexHandler is the handler for root path
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	log.Printf("indexHandler is called\n")
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(appResponse{Message: "OK"})
+}
+
+// this handles /dapr/subscribe, which is called from dapr into this app.
+// this returns the list of topics the app is subscribed to.
+func configureSubscribeHandler(w http.ResponseWriter, r *http.Request) {
+	log.Printf("configureSubscribeHandler called\n")
+
+	var t topicsList
+	t.Topic = append(t.Topic, pubsubA, pubsubB)
+	log.Printf("configureSubscribeHandler subscribing to:%v\n", t.Topic)
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(t.Topic)
+}
+
+// this handles messages published to "pubsub-a-topic"
+func subscribeHandler(w http.ResponseWriter, r *http.Request) {
+	log.Printf("aHandler is called %s\n", r.URL)
+
+	w.WriteHeader(http.StatusOK)
+	defer r.Body.Close()
+
+	var err error
+	var data []byte
+	var body []byte
+	if r.Body != nil {
+		if data, err = ioutil.ReadAll(r.Body); err == nil {
+			body = data
+			log.Printf("assigned\n")
+		}
+	} else {
+		// error
+		err = errors.New("r.Body is nil")
+	}
+
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(appResponse{
+			Message: err.Error(),
+		})
+		return
+	}
+
+	msg, err := extractMessage(body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(appResponse{
+			Message: err.Error(),
+		})
+		return
+	}
+
+	lock.Lock()
+	defer lock.Unlock()
+	if strings.HasSuffix(r.URL.String(), pubsubA) {
+		receivedMessagesA = append(receivedMessagesA, msg)
+	} else if strings.HasSuffix(r.URL.String(), pubsubB) {
+		receivedMessagesB = append(receivedMessagesB, msg)
+	} else {
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(appResponse{
+			Message: err.Error(),
```

Comment:
err is null, you must use your own error here.

Reply:
Good catch.